### PR TITLE
refactor: improve code readability from event-bus

### PR DESCRIPTION
**Pull request by mvares**
Provides more complete documentation and refactors some functions for better `efficiency` and `readability`

### CODE REVIEW

**Comment by carlos-r-l-rodrigues** [comment]
@mgrithm could you please setup your IDE/Prettier to respect the `.eslint` to format the code?

**Comment by mgrithm** [comment]
> @mgrithm could you please setup your IDE/Prettier to respect the `.eslint` to format the code?

Yes. I will send other commit to fix it

File: packages/core/utils/src/event-bus/message-aggregator.ts [hunk 1]
```diff
@@ -14,109 +14,90 @@ export class MessageAggregator implements IMessageAggregator {
     this.messages = []
   }
 
-  save(msg: Message | Message[]): void {
-    if (!msg || (Array.isArray(msg) && msg.length === 0)) {
-      return
-    }
+  save(message: Message | Message[]): void {
+    if (!message || (Array.isArray(message) && message.length === 0)) return
```
Comment (carlos-r-l-rodrigues) [inline packages/core/utils/src/event-bus/message-aggregator.ts:18]:
```suggestion
    if (!message || !message?.length) {
      return
    }
```

File: packages/core/utils/src/event-bus/message-aggregator.ts [hunk 2]
```diff
@@ -14,109 +14,90 @@ export class MessageAggregator implements IMessageAggregator {
...
-      }>((acc, msg) => {
+      messages = messagesCopy.reduce((acc, msg) => {
         const key = groupBy
           .map((field) => this.getValueFromPath(msg, field))
           .join("-")
-        if (!acc[key]) {
-          acc[key] = []
-        }
+
+        acc[key] = acc[key] || []
```
Comment (carlos-r-l-rodrigues) [inline packages/core/utils/src/event-bus/message-aggregator.ts:54]:
```suggestion
        acc[key] ??= []
```

File: packages/core/utils/src/event-bus/message-aggregator.ts [hunk 3]
```diff
@@ -14,109 +14,90 @@ export class MessageAggregator implements IMessageAggregator {
...
 
-  private getValueFromPath(obj: any, path: string): any {
-    const keys = path.split(".")
-    for (const key of keys) {
-      obj = obj[key]
-      if (obj === undefined) break
-    }
-    return obj
+  private getValueFromPath(obj: any, path: string) {
+    return path.split(".").reduce((o, key) => (o ? o[key] : undefined), obj)
```
Comment (carlos-r-l-rodrigues) [inline packages/core/utils/src/event-bus/message-aggregator.ts:69]:
```suggestion
    return path.split(".").reduce((o, key) => o?.[key], obj)
```

File: packages/core/utils/src/event-bus/message-aggregator.ts [hunk 4]
```diff
@@ -14,109 +14,90 @@ export class MessageAggregator implements IMessageAggregator {
...
   }
 
   private compareMessages(
     a: Message,
     b: Message,
     sortBy: MessageAggregatorFormat["sortBy"]
   ): number {
-    for (const key of Object.keys(sortBy!)) {
-      const orderCriteria = sortBy![key]
+    if (!sortBy) return 0
```
Comment (carlos-r-l-rodrigues) [inline packages/core/utils/src/event-bus/message-aggregator.ts:77]:
```suggestion
    if (!sortBy) {
      return 0
    }
```

File: packages/core/utils/src/event-bus/message-aggregator.ts [hunk 5]
```diff
@@ -14,109 +14,90 @@ export class MessageAggregator implements IMessageAggregator {
...
-        } else if (indexA === -1) {
-          return 1
-        } else if (indexB === -1) {
-          return -1
-        } else {
-          return indexA - indexB
-        }
+        if (indexA === indexB) continue
+        else if (indexA === -1) return 1
+        else if (indexB === -1) return -1
```
Comment (carlos-r-l-rodrigues) [inline packages/core/utils/src/event-bus/message-aggregator.ts:90]:
```suggestion
        if (indexA === indexB) {
          continue
        } else if (indexA === -1) {
          return 1
        } else if (indexB === -1) {
          return -1
        }
```

File: packages/core/utils/src/event-bus/message-aggregator.ts [hunk 6]
```diff
@@ -14,109 +14,90 @@ export class MessageAggregator implements IMessageAggregator {
...
-        } else if (valueA < valueB) {
-          return -1 * orderMultiplier
-        } else {
-          return 1 * orderMultiplier
-        }
+        const orderMultiplier =
+          orderCriteria === "desc" || orderCriteria === -1 ? -1 : 1
+
+        if (valueA === valueB) continue
+        else if (valueA < valueB) return -1 * orderMultiplier
```
Comment (carlos-r-l-rodrigues) [inline packages/core/utils/src/event-bus/message-aggregator.ts:98]:
```suggestion
        if (valueA === valueB) {
          continue
        } else if (valueA < valueB) {
          return -1 * orderMultiplier
        }
```

**Review by carlos-r-l-rodrigues — COMMENTED** [review]
I left some small suggestions to match our code formatting.
Thanks for the contribution.

**Pull request closed without merging**